### PR TITLE
Ipamutils: various cleanups and refactor

### PR DESCRIPTION
**Pull request by thaJeztah**
- ipamutils: use RWMutex to allow concurrent reads
- ipamutils: reformat for readability, and improve godoc
- ipamutils: un-export global variables, and use accessors instead
  The `GetLocalScopeDefaultNetworks()` and `GetGlobalScopeDefaultNetworks()` protect these variables with a lock, but various parts in the code were directly accessing the variables without locks.
- ipamutils: generate default pools on first use, instead of init()
  This package defines the `NetworkToSplit` type. Curently importing the package also initializes the default address pools (which may not be needed, e.g., when used in the client.
  This patch changes initialization to happen on first use, instead of `init()`, to allow the package to be imported without initializing these pools.

### CODE REVIEW

**Comment by thaJeztah** [comment]
@arkodg @cpuguy83 ptal

File: ipamutils/utils.go [hunk 1]
```diff
@@ -31,16 +48,22 @@ type NetworkToSplit struct {
 	Size int    `json:"size"`
 }
 
-func init() {
```
Comment (arkodg) [inline ipamutils/utils.go:34]: isn't `init` easier to reason with than `initDefaults` ?

Comment (thaJeztah) [reply ipamutils/utils.go:34]: You mean "using an `init()` function"? Or to rename the `initDefaults` (sync.Once) to `init`? (not sure if the latter works; I suspect `init` is somewhat "reserved"))

**Review by sloppyjuicy — APPROVED** [review]

**Comment by corhere** [comment]
- Superseded by https://github.com/moby/moby/pull/44827

**Pull request closed without merging**